tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -409,4 +409,35 @@ var _ = Describe("odo devfile create command tests", func() {
 			Expect(stdout).To(ContainSubstring("Please use `odo push` command to create the component with source deployed"))
 		})
 	})
+
+	When("there is already a devfile in the directory", func() {
+		BeforeEach(func() {
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
+		})
+		It("should fail with appropriate error when --starter is given to odo create", func() {
+			stderr := helper.Cmd("odo", "create", "--starter", "nodejs-starter", "--context", commonVar.Context, "--project", commonVar.Project).ShouldFail().Err()
+			Expect(stderr).To(ContainSubstring("already has a devfile so you cannot provide a starter"))
+		})
+	})
+
+	When("a devfile is provided which has a starter that has its own devfile", func() {
+		var dirLoc, kconfig, globalConfig string
+		BeforeEach(func() {
+			examplesPath := helper.GetExamplePath()
+			dirLoc = helper.CreateNewContext()
+			kconfig = os.Getenv("KUBECONFIG")
+			globalConfig = os.Getenv("GLOBALODOCONFIG")
+			helper.CopyFile(kconfig, filepath.Join(dirLoc, "config"))
+			helper.CopyFile(globalConfig, filepath.Join(dirLoc, "preference.yaml"))
```

Comment:
I think this code is not necessary anymore, now that the kubeconfig and odoconfig files are in a separate directory: 
https://github.com/redhat-developer/odo/blob/main/tests/helper/helper_generic.go#L300
https://github.com/redhat-developer/odo/blob/main/tests/helper/helper_generic.go#L303

Reply:
ok removing